### PR TITLE
Checkout Page - Taxjar Integration 

### PR DESCRIPTION
**Pull request by george-icarcover**
- Make sure tax is calculated after shipping details have been obtained
- Make sure payment routes also show the correct total with tax
- https://app.clickup.com/t/86890zgn8

### CODE REVIEW

File: src/app/(noFooter)/checkout/utils/index.ts [hunk 1]
```diff
@@ -170,30 +179,13 @@ export async function paypalCaptureOrder(orderID: string, phone: string) {
...
+      throw new Error('Network response was not ok');
     }
-
     const data = await response.json();
+
     return data;
   } catch (err) {
-    console.error(`[PaypalCaptureOrder] Error: `, err);
-    throw err;
+    console.error(err);
```
Comment (devwilliamy) [inline src/app/(noFooter)/checkout/utils/index.ts:188]: Why is the error being removed? Should keep this as is

File: src/components/checkout/CheckoutAccordion.tsx [hunk 2]
```diff
@@ -330,6 +329,7 @@ export default function CheckoutAccordion() {
       2
     );
     const totalWithTax = convertPriceToStripeFormat(taxSum);
+    console.log({ amount_to_collect, totalWithTax, taxSum, taxRes });
```
Comment (devwilliamy) [inline src/components/checkout/CheckoutAccordion.tsx:332]: Remove console logs throughout these PR please

File: src/components/checkout/PayPalButtonSection.tsx [hunk 3]
```diff
@@ -27,16 +27,16 @@ import { generateSkuLabOrderInput } from '@/lib/utils/skuLabs';
...
-type PaypalButtonSectionProps = {
-  setPaypalSuccessMessage: (message: string) => void;
-  setMessage: (message: string) => void;
-};
-export default function PayPalButtonSection({
-  setPaypalSuccessMessage,
-  setMessage,
-}: PaypalButtonSectionProps) {
-  const { orderNumber, shipping, shippingAddress, customerInfo } =
-    useCheckoutContext();
```
Comment (devwilliamy) [inline src/components/checkout/PayPalButtonSection.tsx:39]: Somethings not right, why are these removed?

File: src/components/checkout/PayPalButtonSection.tsx [hunk 4]
```diff
@@ -304,19 +295,6 @@ export default function PayPalButtonSection({
...
-              console.error(
-                `[PaypalButtonSection] Error: Popup closed unexpectedly`
-              );
-              setMessage('Popup closed unexpectedly');
-            } else if (error?.message) {
-              setMessage(error?.message);
-            } else {
-              setMessage('Unexpected Error Occurred');
-            }
-          }}
```
Comment (devwilliamy) [inline src/components/checkout/PayPalButtonSection.tsx:319]: This should be in here and not deleted i'm not sure why this is getting taken out

File: src/components/checkout/PayPalButtonSection.tsx [hunk 5]
```diff
@@ -88,21 +90,14 @@ export default function PayPalButtonSection({
...
-            try {
-              response = await paypalCaptureOrder(
-                data.orderID,
-                customerInfo.phoneNumber
-              );
-              // console.log('Response:', response);
-            } catch (error) {
-              // debugger;
-              throw error;
-            }
```
Comment (devwilliamy) [inline src/components/checkout/PayPalButtonSection.tsx:105]: This should be in here

File: src/components/checkout/PayPalButtonSection.tsx [hunk 6]
```diff
@@ -136,10 +131,6 @@ export default function PayPalButtonSection({
             // });
 
             if (response.success) {
-              setMessage('');
-              setPaypalSuccessMessage(
-                'Paypal Payment Accepted, please wait for the page to finish loading.'
-              );
```
Comment (devwilliamy) [inline src/components/checkout/PayPalButtonSection.tsx:142]: This should be in here

**Review by devwilliamy — CHANGES_REQUESTED** [review]
There are some paypal changes that should be in here but are being removed

**Comment by devwilliamy** [comment]
I wasn't able to continue to order review.

Tax is also not showing up..? Not sure where this is supposed to be populated

image

**Comment by george-icarcover** [comment]
> I wasn't able to continue to order review.
> 
> Tax is also not showing up..? Not sure where this is supposed to be populated
> 
> image

If this is before the shipping details have been saved, it will not calculate tax as the taxjar api requires country, state, and zipcode

**Comment by devwilliamy** [comment]
> > I wasn't able to continue to order review.
> > Tax is also not showing up..? Not sure where this is supposed to be populated
> > image
> 
> If this is before the shipping details have been saved, it will not calculate tax as the taxjar api requires country, state, and zipcode

This is at the very end 

image

**Review by devwilliamy — COMMENTED** [review]
I'll need to look at the code again once you've merged in preview again

Something I'm wondering about is logging the tax in the orders table
 
If each item has a tax amount on it, then will need to track in order items table too, but if it's just a tax on the whole order, probably just need a column in the _Orders table to make sure the total amount makes sense with the discounts and other prices

**Comment by devwilliamy** [comment]
Completed in 
Implemented Taxjar and Added Payment Processing Messages #691

**Pull request closed without merging**